test/storageAPI_UnitsForUsers_test.dart

```diff
@@ -0,0 +1,291 @@
+import 'package:protos/storage-api.dart' as storage_api;
+import 'package:flutter_ics_homescreen/export.dart';
+
+import 'package:flutter_ics_homescreen/data/data_providers/storage_client.dart';
+import 'package:flutter_ics_homescreen/data/data_providers/initializeSettings.dart';
+
+
+// Mock implementation of Ref if necessary
+class MockRef extends Ref {
+  @override
+  dynamic noSuchMethod(Invocation invocation) => super.noSuchMethod(invocation);
+}
+
+void main() {
+  late StorageClient storageClient;
+  late ProviderContainer container;
+  
+  setUp(() {
+    storageClient = StorageClient(
+      config: StorageConfig.defaultConfig(),
+      ref: MockRef(),
+    );
+    container = ProviderContainer();
+    //dispose container after each test
+    addTearDown(container.dispose);
+  });
+
+  test('add User', () async {
+    await storageClient.destroyDB();
+    final userClient = container.read(usersProvider.notifier);
+    await userClient.addUser('Mark');
+    //access state
+    var userState = container.read(usersProvider);
+    final userId = userState.users[0].id;
+    final searchResponse = await storageClient.read(storage_api.Key(key: '${VSSPath.vehicleUsers}.$userId.name'));
+    expect(searchResponse.success, isTrue);
+    expect(searchResponse.result, 'Mark');
+    await storageClient.destroyDB();
+  });
+
+  test('remove User', () async{
+    await storageClient.destroyDB();
+    //add User
+    final userClient = container.read(usersProvider.notifier);
+    await userClient.addUser('Mark');
+    //access state
+    var userState = container.read(usersProvider);
+    final markId = userState.users[0].id;
+
+    //remove User
+    await userClient.removeUser(markId);
+    final searchResponse = await storageClient.search(storage_api.Key(key: markId));
+    expect(searchResponse.success, isTrue);
+    expect(searchResponse.result, []);
+    await storageClient.destroyDB();
+  });
+
+  test('add users, select user', () async{
+    await storageClient.destroyDB();
+    //add Users
+    final userClient = container.read(usersProvider.notifier);
+    await userClient.addUser('Mark');
+    await userClient.addUser('Clara');
+
+    var userState = container.read(usersProvider);
+    final markId = userState.users[1].id;
+    await userClient.selectUser(markId);
+
+    final readResponseAfterSelectUser = await storageClient.read(storage_api.Key(key: VSSPath.vehicleCurrentUser));
+    expect(readResponseAfterSelectUser.success, isTrue);
+    expect(readResponseAfterSelectUser.result, markId);
+    await storageClient.destroyDB();
+  });
+
+    test('selected user default', () async{
+    await storageClient.destroyDB();
+
+    //access state
+    var userState = container.read(usersProvider);
+    final selectedId = userState.selectedUser.id;
+
+    final readResponse = await storageClient.read(storage_api.Key(key: VSSPath.vehicleCurrentUser));
+    expect(readResponse.success, isFalse);
+    expect(selectedId, '');
```

Comment:
is this still up to date? i thought we replaced the default users and gave them more meaningful ids?